### PR TITLE
Add Bicep templates and extend deployment pipeline

### LINKED ISSUE #18: Include infrastructure in code
As DevOps engineer, I should use infrastructure as code to create environments for deployment. Utilize templates based on target environment (Bicep). Setup infrastructure deployment as part of CI pipeline.

### PR DESCRIPTION
**Pull request by uveta**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -28,31 +27,79 @@ permissions:
   pull-requests: write
 
 jobs:
-  build_and_deploy:
+  build:
     if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.action != 'closed')
```
Comment (vuleips) [inline .github/workflows/ci.yaml:31]: this condition check is reused for some later job.
how about storing it to variable?

**Review by vuleips — APPROVED** [review]
The PR looks good to me.
I like the way you setup link between Azure deployment to Git Workflow run (so it's easier to track).
To enable the SWA to have Staging + Production is also great. So automatic CI only deploy changes to Staging.

I have a question:
How the deploy to Production is made? It is a manual approval via Azure Portal?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (uveta) [reply .github/workflows/ci.yaml:31]: I tried, and there doesnt seem to be a context available, where we can set and access such variable. We could set it in **env** context, but cannot use it. For job conditionals, only **github** context is available, which is read only :(

**Comment by uveta** [comment]
> The PR looks good to me. I like the way you setup link between Azure deployment to Git Workflow run (so it's easier to track). To enable the SWA to have Staging + Production is also great. So automatic CI only deploy changes to Staging.
> 
> I have a question: How the deploy to Production is made? It is a manual approval via Azure Portal?

Production deployment is done whenever **main** branch is updated. Handling different environments is done automatically by Azure Static Web Apps, so we dont have to worry about it.

**Pull request merged**